### PR TITLE
[PB-1435]: feat/update users tier on subscription change.

### PR DESCRIPTION
**Pull request by apsantiso**
With the implementation of the new feature limits. We need to update users's tiers based on their subscription status, in order to accomplish that, we need to use the new /update/tier endpoint that is exposed on the gateway.

Requires:

- https://github.com/internxt/drive-server/pull/427
- https://github.com/internxt/drive-server/pull/426 (merged)

NOTE: this should NOT be merged without mapping the productIds/planIds - tiersId in paid_plans beforehand!!

### CODE REVIEW

**Comment by sw-wayner** [comment]
/update-preview drive-server-wip preview-274-31558bb7d55d2271e84a32c00d03da446f0ad967 imageSuffix=-dev

**Comment by inxtci** [comment]
> [!NOTE]
> Pull request successfully updated 🚀
> Preview updated with drive-server-wip-dev:preview-274-31558bb7d55d2271e84a32c00d03da446f0ad967

**Comment by apsantiso** [comment]
/update-preview drive-server preview-427-f2e81c3c78c92f8377a0f3b270fed63fcecd072b imageSuffix=-dev

**Comment by inxtci** [comment]
> [!NOTE]
> Pull request successfully updated 🚀
> Preview updated with drive-server-dev:preview-427-f2e81c3c78c92f8377a0f3b270fed63fcecd072b

**Comment by sw-wayner** [comment]
/update-preview drive-server preview-427-f2e81c3c78c92f8377a0f3b270fed63fcecd072b imageSuffix=-dev

**Comment by sw-wayner** [comment]
/update-preview drive-server-wip preview-274-31558bb7d55d2271e84a32c00d03da446f0ad967 imageSuffix=-dev

**Comment by inxtci** [comment]
> [!CAUTION]
> An error has ocurred processing the command arguments
> **Tag argument not found**

**Comment by sw-wayner** [comment]
/update-preview drive-server-wip preview-274-31558bb7d55d2271e84a32c00d03da446f0ad967 imageSuffix=-dev

**Comment by sw-wayner** [comment]
/update-preview drive-server preview-427-f2e81c3c78c92f8377a0f3b270fed63fcecd072b imageSuffix=-dev

**Comment by sw-wayner** [comment]
/update-preview notifications preview-20-825e3d6706fa8cb81b817ff27f331c00812c8f59 imageSuffix=-dev

**Comment by sw-wayner** [comment]
/update-preview drive-server-wip preview-274-31558bb7d55d2271e84a32c00d03da446f0ad967 imageSuffix=-dev

File: src/webhooks/handleCheckoutSessionCompleted.ts [hunk 1]
```diff
@@ -59,6 +59,17 @@ export default async function handleCheckoutSessionCompleted(
     throw err;
   }
 
+  try {
+    await updateUserTier(user.uuid, price.product as string, config);
+  } catch (err) {
+    log.error(
+      `Error while updating user tier: email: ${session.customer_email}, planId: ${price.product} `,
+    );
+    log.error(err);
```
Comment (SimonGodefroid) [inline src/webhooks/handleCheckoutSessionCompleted.ts:68]: can't say for sure but more often than not logging systems logs `[object Object]` unless you 
```tsx
JSON.stringify(err,null,'\t')
```

**Comment by apsantiso** [comment]
/update-preview drive-server-wip preview-20ec4607ea04b0ae2c46d9fd7bc389dd1fd2b6c2 imageSuffix=-dev

**Comment by apsantiso** [comment]
/update-preview drive-server-wip 20ec4607ea04b0ae2c46d9fd7bc389dd1fd2b6c2 imageSuffix=-dev

**Comment by apsantiso** [comment]
/update-preview drive-server-wip preview-274-c2c3a96e3ef400219b757266e3dd352be29f2874 imageSuffix=-dev

**Comment by inxtci** [comment]
> [!CAUTION]
> Pull request update failed with following error:
> **error: unable to find container named "payments"**

**Comment by inxtci** [comment]
> [!CAUTION]
> Pull request not updated after 5 minutes 💥
> We tried to update the image with payments-dev:preview-52-a1ce7ca75a0f3919cc51b61d69978f22ebb99c03 
> If the arguments are correct, please check the logs. Or try again with another deployment, tag, or imageSuffix configuration.

**Comment by inxtci** [comment]
> [!CAUTION]
> Pull request update failed with following error:
> **error: unable to find container named "payments"**

**Comment by inxtci** [comment]
> [!CAUTION]
> Pull request not updated after 5 minutes 💥
> We tried to update the image with payments-dev:preview-52-a1ce7ca75a0f3919cc51b61d69978f22ebb99c03 
> If the arguments are correct, please check the logs. Or try again with another deployment, tag, or imageSuffix configuration.

**Comment by apsantiso** [comment]
/update-preview drive-server-wip preview-274-c2c3a96e3ef400219b757266e3dd352be29f2874 imageSuffix=-dev

**Comment by inxtci** [comment]
> [!CAUTION]
> Pull request update failed with following error:
> **error: unable to find container named "payments"**

**Comment by inxtci** [comment]
> [!CAUTION]
> Pull request not updated after 5 minutes 💥
> We tried to update the image with payments-dev:preview-52-a1ce7ca75a0f3919cc51b61d69978f22ebb99c03 
> If the arguments are correct, please check the logs. Or try again with another deployment, tag, or imageSuffix configuration.

**Comment by inxtci** [comment]
> [!CAUTION]
> Pull request update failed with following error:
> **error: unable to find container named "payments"**

**Comment by inxtci** [comment]
> [!CAUTION]
> Pull request not updated after 5 minutes 💥
> We tried to update the image with payments-dev:preview-52-f5021f01b9531363383a48651f299f522d9ce436 
> If the arguments are correct, please check the logs. Or try again with another deployment, tag, or imageSuffix configuration.

**Comment by apsantiso** [comment]
/update-preview drive-server-wip preview-274-c2c3a96e3ef400219b757266e3dd352be29f2874 imageSuffix=-dev

**Comment by apsantiso** [comment]
/update-preview drive-server preview-427-c1e69124a1fe2795c613dedd0dfe28f51151609f imageSuffix=-dev

**Review by sg-gs — APPROVED** [review]

**Pull request merged**